### PR TITLE
Added .clang-format file, ran it over the codebase

### PR DESCRIPTION
**Pull request by larskraemer**
I added a .clang-format file to the project and let it format everything. This should keep the style consistent. The style is based on the built-in Microsoft style, as this matched the existing style the closest. Let me know if you'd like me to change some of the settings.

### CODE REVIEW

File: Url.cpp [hunk 1]
```diff
@@ -141,100 +148,98 @@ std::string Url::encode(const std::string& str)
...
     }
 }
 
 bool Url::parse()
 {
-    std::string_view url_view {url_string};
+    std::string_view url_view{url_string};
 
-    auto current {url_view.find("://")};
+    auto current{url_view.find("://")};
```
Comment (ameenmaali) [inline Url.cpp]: Can we maintain this space? Not a fan of how it is currently

File: Url.cpp [hunk 2]
```diff
@@ -141,100 +148,98 @@ std::string Url::encode(const std::string& str)
...
-
 void Url::regex_parse()
 {
     std::smatch match;
     if (std::regex_match(url_string, match, URL_REGEX))
     {
-        scheme = match[2].str() + "://";
-        hostname = match[4];
-        path = match[5];
+        scheme        = match[2].str() + "://";
```
Comment (ameenmaali) [inline Url.cpp]: Can we keep this how it was? Seems odd to do this for normal variables - I would understand for something like a struct or similar

File: Url.cpp [hunk 3]
```diff
@@ -141,100 +148,98 @@ std::string Url::encode(const std::string& str)
...
         qs.erase(0, current + 1);
 
-        std::string qs_key {token.substr(0, token.find('='))};
+        std::string qs_key{token.substr(0, token.find('='))};
         qs_vals.push_back(qs_key);
     }
 
     url_key += "?";
-    for (const auto &x: qs_vals)
+    for (const auto &x : qs_vals)
```
Comment (ameenmaali) [inline Url.cpp]: Can we keep this one how it was?

**Review by ameenmaali — APPROVED** [review]
Thanks @larskraemer! Never hurts to have a clean style, even for smaller projects :) - I had a few comments on things I'd like to keep how they were, but other than that it looks good to me!

**Comment by larskraemer** [comment]
Alright, these are the changes: 
`SpaceBeforeCpp11BracedList` is now `true`
`SpaceBeforeRangeBasedForLoopColon` is now `false`
`AlignConsecutiveAssignments` is now `false` (honestly, i kind of liked it)

**Comment by ameenmaali** [comment]
Thanks!

**Pull request merged**